hub/core/storage/s3.py

```diff
@@ -18,7 +18,7 @@ def __init__(
         aws_session_token: Optional[str] = None,
         endpoint_url: Optional[str] = None,
         aws_region: Optional[str] = None,
-        max_pool_connections: Optional[int] = 10,
+        max_pool_connections: Optional[int] = 50,
```

Comment:
Might need to increase the default even further, I've been using up to 108 workers till now.
Would there be any problems with having this number too high?